Python/PRP/GUI/pyGUIDialogMod.cpp

```diff
@@ -0,0 +1,117 @@
+/* This file is part of HSPlasma.
+*
+* HSPlasma is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* HSPlasma is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with HSPlasma.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <PRP/GUI/pfGUIDialogMod.h>
+#include "pyGUIDialogMod.h"
+#include "pyGUIControlMod.h"
+#include "PRP/Modifier/pyModifier.h"
+#include "PRP/KeyedObject/pyKey.h"
+
+extern "C" {
+
+PY_PLASMA_NEW(GUIDialogMod, pfGUIDialogMod)
+
+PY_METHOD_VA(GUIDialogMod, addControl,
+    "Params: control\n"
+    "Add a control to the dialog mod")
+{
+    pyKey* control;
+    if (!PyArg_ParseTuple(args, "O", &control)) {
+        PyErr_SetString(PyExc_TypeError, "addControl expects a plKey");
+        return NULL;
+    }
+    if (!pyKey_Check((PyObject*)control)) {
+        PyErr_SetString(PyExc_TypeError, "addControl expects a plKey");
+        return NULL;
+    }
+    self->fThis->addControl(*control->fThis);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_VA(GUIDialogMod, delControl,
+    "Params: idx\n"
+    "Remove a control from the dialog mod")
+{
+    int idx;
+    if (!PyArg_ParseTuple(args, "i", &idx)) {
+        PyErr_SetString(PyExc_TypeError, "delControl expects an int");
+        return NULL;
+    }
+    self->fThis->delControl(idx);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_NOARGS(GUIDialogMod, clearControls,
+    "Remove all controls from the dialog mod")
+{
+    self->fThis->clearControls();
+    Py_RETURN_NONE;
+}
+
+PY_GETSET_GETTER_DECL(GUIDialogMod, controls) {
+    PyObject* list = PyTuple_New(self->fThis->getControls().size());
+    for (size_t i = 0; i < self->fThis->getControls().size(); i++)
+        PyTuple_SET_ITEM(list, i, pyPlasma_convert(self->fThis->getControls()[i]));
+    return list;
+}
+
+static PyMethodDef pyGUIDialogMod_Methods[] = {
+    pyGUIDialogMod_addControl_method,
+    pyGUIDialogMod_delControl_method,
+    pyGUIDialogMod_clearControls_method,
+    PY_METHOD_TERMINATOR
+};
+
+PY_PROPERTY(int, GUIDialogMod, tagID, getTagID, setTagID)
+PY_PROPERTY(int, GUIDialogMod, version, getVersion, setVersion)
+PY_PROPERTY(plKey, GUIDialogMod, renderMod, getRenderMod, setRenderMod)
+PY_PROPERTY(ST::string, GUIDialogMod, name, getName, setName)
+PY_PROPERTY(plKey, GUIDialogMod, procReceiver, getProcReceiver, setProcReceiver)
+PY_PROPERTY(plKey, GUIDialogMod, sceneNode, getSceneNode, setSceneNode)
+PY_PROPERTY(pfGUIColorScheme, GUIDialogMod, colorScheme, getColorScheme, setColorScheme)
```

Comment:
Should use a `PY_PROPERTY_PROXY`, so the setter isn't needed (we're only modifying the stored colorScheme, not creating a new object).